ls1intum/Artemis
# Plagiarism check for modeling exercises

### Checklist

### Motivation and Context
Instructors should be able to check plagiarism for modeling exercises

### Description
* Add new endpoint that reports the results in the JSON response (this will be enhanced in the future)
* Use the existing similarity checks as part of Compass for class and activity diagrams
* Add support for similarity checks of UML communication diagrams
* Additional diagram types (use case, deployment, component) will be implemented in a follow up PR.

### Steps for Testing
1. Create a modeling exercises
2. Submit 2 identical models from different students
3. Navigate to the modeling exercise --> view --> Check plagiarism
4. Look into the Chrome network tab and check that the response includes a similarity of 1.0
